### PR TITLE
Github Actions Workflows

### PR DESCRIPTION
**Pull request by therahulbhati**
- [x] Workflow for on push and pull request to master PR

---
This change is [Reviewable](https://reviewable.io/reviews/fission/fission/1876)

### CODE REVIEW

File: test/tests/test_environments/test_go_env.sh [hunk 1]
```diff
@@ -70,7 +70,7 @@ fission fn update --name $fn_poolmgr --pkg $pkgName
 fission fn update --name $fn_nd --pkg $pkgName
 
 log "Waiting for router & pools to catch up"
-sleep 5
```
Comment (vishal-biyani) [inline test/tests/test_environments/test_go_env.sh:73]: 300 is quite a lot, did we try with smaller numbers?

File: test/tests/test_ingress.sh [hunk 2]
```diff
@@ -1,4 +1,6 @@
 #!/bin/bash
+#test:disabled
```
Comment (vishal-biyani) [inline test/tests/test_ingress.sh:2]: Would be good to add a small note that this is disabled only for Kind tests and for GKE/EKS tests should be enabled (Once we get that framework in place)

**Comment by therahulbhati** [comment]
Raising from another branch.

File: test/utils.sh [hunk 3]
```diff
@@ -170,25 +174,25 @@ waitBuildExpectedStatus() {
 export -f waitBuildExpectedStatus
 
 
-## Common env parameters
-export FISSION_NAMESPACE=${FISSION_NAMESPACE:-fission}
-export FUNCTION_NAMESPACE=${FUNCTION_NAMESPACE:-fission-function}
+# ## Common env parameters
+# export FISSION_NAMESPACE=${FISSION_NAMESPACE:-fission}
```
Comment (vishal-biyani) [inline test/utils.sh:178]: Will we be using this later for other tests?

**Review by vishal-biyani — CHANGES_REQUESTED** [review]
Overall looks pretty good @therahulbhati - very minor comments to be fixed and we are good to go!

File: go.mod [hunk 4]
```diff
@@ -28,22 +27,19 @@ require (
 	github.com/go-ini/ini v1.57.0 // indirect
 	github.com/go-openapi/spec v0.17.2
 	github.com/golang/freetype v0.0.0-20170609003504-e2365dfdc4a0 // indirect
-	github.com/golang/groupcache v0.0.0-20190129154638-5b532d6fd5ef // indirect
```
[the remark this answers is not in the document]
Comment (therahulbhati) [reply go.mod:31]: Yes, there is some dependency problem introduced due to change, I have opened a PR for same https://github.com/fission/fission/pull/1879. It is necessary for this PR else creation of fission images will fail

File: kind.yaml [hunk 5]
```diff
@@ -23,7 +23,7 @@ nodes:
...
     protocol: TCP
-- role: worker
-  image: kindest/node:v1.16.4
-- role: worker
-  image: kindest/node:v1.16.4
\ No newline at end of file
+# - role: worker
+#   image: kindest/node:v1.16.4
+# - role: worker
+#   image: kindest/node:v1.16.4
```
[the remark this answers is not in the document]
Comment (therahulbhati) [reply kind.yaml]: Removed

File: skaffold.yaml [hunk 6]
```diff
@@ -65,3 +66,17 @@ profiles:
       - op: replace
         path: /deploy/helm/releases/0/setValues/routerServiceType
         value: "NodePort"
+  - name: kind-CI
```
[the remark this answers is not in the document]
Comment (therahulbhati) [reply skaffold.yaml]: Done

File: test/tests/test_ingress.sh (hunk 2 above)
Comment (therahulbhati) [reply test/tests/test_ingress.sh]: Added note

File: test/utils.sh (hunk 3 above)
Comment (therahulbhati) [reply test/utils.sh]: No, it's not required, so removed.

File: test/tests/test_environments/test_go_env.sh (hunk 1 above)
Comment (therahulbhati) [reply test/tests/test_environments/test_go_env.sh]: Done

File: .github/workflows/push_pr.yaml [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+on: 
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  # Job to run change detection
+  buildenv:
```
Comment (vishal-biyani) [inline .github/workflows/push_pr.yaml:12]:
```suggestion
  integration-test:
```

**Comment by vishal-biyani** [comment]
Thanks a lot @agiwalpooja20 @abhisheks-infracloud - who initiated this work, @sahil-lakhwani who took more steps, and finally @therahulbhati for closing this. This is a milestone in moving our CI to Github Actions, thanks for all your efforts 🎉 💯

**Comment by abhisheks-infracloud** [comment]
Awesome !!

**Pull request merged**